app/src/sharedTest/java/org/oppia/app/testing/InputInteractionViewTestActivityTest.kt

```diff
@@ -215,11 +207,208 @@ class InputInteractionViewTestActivityTest {
+    onView(withId(R.id.test_number_with_units_input_interaction_view)).perform(typeText("5 9/10 km"))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_with_units_input_interaction_view) as NumberWithUnitsInputInteractionView
+      val numberWithUnits = textAnswerRetriever.getPendingAnswer().numberWithUnits
+      assertThat(numberWithUnits.fraction.isNegative).isEqualTo(false)
+      assertThat(numberWithUnits.fraction.wholeNumber).isEqualTo(5)
+      assertThat(numberWithUnits.fraction.numerator).isEqualTo(9)
+      assertThat(numberWithUnits.fraction.denominator).isEqualTo(10)
+      assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("km")
+    }
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withInputtedValueUnit_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_number_with_units_input_interaction_view)).perform(typeText("100 Rs."))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_with_units_input_interaction_view) as NumberWithUnitsInputInteractionView
+      val numberWithUnits = textAnswerRetriever.getPendingAnswer().numberWithUnits
+      assertThat(numberWithUnits.fraction.wholeNumber).isEqualTo(100)
+      assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("Rs.")
+    }
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withInputtedDecimalValueAndUnit_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_number_with_units_input_interaction_view)).perform(typeText("100.50 Km"))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_with_units_input_interaction_view) as NumberWithUnitsInputInteractionView
+      val numberWithUnits = textAnswerRetriever.getPendingAnswer().numberWithUnits
+      assertThat(numberWithUnits.real).isEqualTo(100.50f)
+      assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("Km")
+    }
+  }
+
+  @Test
+  fun testNumberWithUnitsInputInteractionView_withInputtedWithSymbols_hasCorrectPendingAnswer() {
+    val activityScenario = ActivityScenario.launch(InputInteractionViewTestActivity::class.java)
+    onView(withId(R.id.test_number_with_units_input_interaction_view)).perform(typeText("100.50 Km/m^2"))
+    activityScenario.onActivity { activity ->
+      val textAnswerRetriever =
+        activity.findViewById(R.id.test_number_with_units_input_interaction_view) as NumberWithUnitsInputInteractionView
+      val numberWithUnits = textAnswerRetriever.getPendingAnswer().numberWithUnits
+      assertThat(numberWithUnits.real).isEqualTo(100.50f)
+      assertThat(numberWithUnits.getUnit(0).unit).isEqualTo("Km/m^2")
```

Comment:
I think this isn't correct. We should be capturing the exponent and setting it as '2' here.